src/mimeparser.rs

```diff
@@ -1302,6 +1303,18 @@ fn get_attachment_filename(mail: &mailparse::ParsedMail) -> Result<Option<String
     let desired_filename =
         desired_filename.or_else(|| mail.ctype.params.get("name").map(|s| s.to_string()));
 
+    // decode filename as CHARSET'LANG'test%2E%70%64%66.
+    // we're always assuming utf-8,
+    // however, due to lossy decoding, the extension is preserved
+    // and things should not be totally bad for other encodings.
+    // we can tweak that when we see sth. else really used in the wild nowadays.
+    let desired_filename = if let Some(name) = desired_filename {
+        let name = name.splitn(3, '\'').last().unwrap().to_string();
```

Comment:
It is probably true that `SplitN` iterator returns at least one item, but I'd use `if let` anyway and return `None` in the `else` branch. Rust panics are hard to debug when they crash the app on mobile.